File: client/tpnrp-ui/src/types/inventory.ts
export type TInventoryItem = {
    amount: number
    name: string
    label: string
    weight: number // Weight in gram
    slot: number // slot number of this item
    description?: string // Description of this item
    useable?: boolean // Whether this item is useable
    unique?: boolean // Whether this item is unique (cannot stack)
    info?: {
        rare?: number // From 1 to 5
        durability?: number // From 0 to 100
        maxWeight?: number // In gram
        slot?: number // slot count of this item (This field is for backpack)
    }
}

export type TInventoryGroup = 'equipment' | 'inventory' | 'container' | 'devLibrary'

export type TInventoryItemProps = {
    item?: TInventoryItem
    slot?: number
    group?: TInventoryGroup
    isShowHotbarNumber?: boolean
    isDragDropDisabled?: boolean
}

export type TCraftingRecipe = {
    
}

export type TInventoryOpenInventoryResultType = {
    status: boolean
    message: string
    inventory: TInventoryItem[]
    container: {
        id: string
        items: TInventoryItem[]
    } | null
}

export type TResponseCreateDropItem = {
    status: boolean
    message: string
    itemData: {
        itemName: string
        amount: number
        fromSlot: number
    }
}

export type TResponseSplitItem = {
    status: boolean
    message: string
}

File: shared/types/inventory.lua
---@class SInventoryType
---@field id string
---@field citizen_id string
---@field inventories table<number, SInventoryItemType>
---@field type 'player' | 'stack'

---@class SInventoryItemType
---@field amount number
---@field info table
---@field slot number
---@field name string           -- [DB]
---@field label string          -- [DB]
---@field weight number         -- [DB]
---@field type string           -- [DB] 'item' | 'weapon' | 'ammo' | 'tool' | 'material' | 'other'
---@field image string          -- [DB]
---@field unique boolean        -- [DB] this item is unique for slot. Meaning 1 slot can only have 1 of this item
---@field useable boolean       -- [DB]
---@field shouldClose boolean   -- [DB]
---@field description string    -- [DB]

---@class SInventoryCanAddItemResultType
---@field status boolean
---@field message string

---@class SInventoryAddItemResultType
---@field status boolean
---@field message string
---@field slot number

---@class TContainer
---@field id string -- Id of container
---@field items table<number, SInventoryItemType> -- list items in container
---@field entityId string -- Id of entity
---@field entity unknown -- Entity of container
---@field maxSlot number -- Max slot count
---@field maxWeight number -- Max weight in grams

---@class TInventoryOpenInventoryResultType -- Response object when called open inventory
---@field status boolean -- Status of response
---@field message string -- message
---@field inventory table<number, SInventoryItemType> -- player's inventory
---@field equipment table<EEquipmentClothType, SEquipmentItemType> -- player's equipment
---@field container {id: string; items: table<number, SInventoryItemType>}|nil -- container's inventory

---@class TResponseCreateDropItem
---@field status boolean -- Status of response
---@field message string -- message
---@field itemData { itemName: string; amount: number; fromSlot: number }


---@class TWorldItem
---@field path string   - Path to origin resource
---@field scale Vector3 - Scale of this entity in world